src/Microsoft.AspNet.Mvc.Core/FilterActionInvoker.cs

```diff
@@ -654,24 +654,9 @@ private IFilterMetadata[] GetFilters()
                 {
                     await item.FilterAsync.OnResultExecutionAsync(_resultExecutingContext, InvokeResultFilterAsync);
 
-                    if (_resultExecutedContext == null)
+                    if (_resultExecutedContext == null || _resultExecutingContext.Cancel == true)
                     {
-                        // Short-circuited by not calling next
-
-                        _logger.LogVerbose(ResourceFilterShortCircuitLogMessage, item.FilterAsync.GetType().FullName);
-
-                        _resultExecutedContext = new ResultExecutedContext(
-                            _resultExecutingContext,
-                            _filters,
-                            _resultExecutingContext.Result,
-                            Instance)
-                        {
-                            Canceled = true,
-                        };
-                    }
-                    else if (_resultExecutingContext.Cancel == true)
-                    {
-                        // Short-circuited by setting Cancel == true
+                        // Short-circuited by not calling next || Short-circuited by setting Cancel == true
```

Comment:
If these blocks really have the same code we should look at cleaning that up across the board, there are probably 2-3 more cases in the invoker.